xscen/utils.py

```diff
@@ -39,8 +45,76 @@
     "translate_time_chunk",
     "unstack_fill_nan",
     "unstack_dates",
+    "update_attr",
 ]
 
+TRANSLATOR = defaultdict(lambda: lambda s: s)
+"""Dictionary of translating objects.
+
+Each key is a two letter locale code and values are functions that return the translated message as compiled in the gettext catalogs.
+If a language is not defined or a message not translated, the function will return the raw message.
+"""
+
+
+for loc in (Path(__file__).parent / "data").iterdir():
+    if loc.is_dir() and len(loc.name) == 2:
+        TRANSLATOR[loc.name] = gettext.translation(
+            "xscen", localedir=loc.parent, languages=[loc.name]
+        ).gettext
+
+
+def update_attr(ds, attr, new, others=None, **fmt):
+    """Format an attribute referencing itself in a translatable way.
+
+    Parameters
+    ----------
+    ds: Dataset or DataArray
+        The input object with the attribute to update.
+    attr : str
+        Attribute name.
+    new : str
+        New attribute as a template string. It may refer to the old version
+        of the attribute with the "{attr}" field.
+    others: iterable of Datasets or DataArrays
+        Other objects from which we can extract the attributes.
+        The attributes can be be referenced as "{attrII}" in `new`, where II
+        is the index of the other source.
```

Comment:
not sure what you mean by "the index of the other source". 
Also, I might be a bit more explicit that we are getting the attribute `attrs` from this new object.